design/ceph/object/admin-ops.md

```diff
@@ -0,0 +1,57 @@
+---
+title: deploy a separate RGW instance to serve admin-ops API
+target-version: release-1.16
+---
+
+# Feature Name
+Separate RGW deployment to serve admin-ops API.
+
+## Summary
+
+RGW provides an [admin-ops API](https://docs.ceph.com/en/latest/radosgw/adminops/) to manage users, usage, quotas, and other sensitive information. This API is hosted on the same port as the object storage APIs (S3 and Swift), meaning that object store users have physical access to the Admin API, which is only secured by authorization controls.
+
+This configuration presents a security risk: obtaining access to admin credentials could allow unauthorized access to the Admin API. A more secure approach would be to hide the Admin API behind a firewall. The most straightforward solution would involve hosting the Admin API on a separate port; however, [this feature](https://tracker.ceph.com/issues/68484) is not yet implemented in RGW.
+
+Currently, RGW offers the [`rgw_enable_apis` option](https://docs.ceph.com/en/reef/radosgw/config-ref/#confval-rgw_enable_apis), which allows disabling the Admin API for a given RGW instance. However, the Admin API cannot be entirely disabled, as Rook relies on it for critical operations.
+
+The proposed solution involves disabling the Admin Ops API on user-facing RGW instances and deploying a separate RGW instance with only the Admin Ops API enabled. This separate deployment would allow configuring a dedicated Kubernetes service and using a separate domain for the Admin API, enhancing overall security.
+
+## Proposal details
+
+Add admin ops RGW instance options to CephObjectStore CRD:
+
+```diff yaml
+apiVersion: ceph.rook.io/v1
+kind: CephObjectStore
+metadata:
+  name: my-store
+spec:
+...
++  # Deploys a separate RGW instance to host admin ops API
++  extAdminOps:
++    enabled: true
++    port: 80
++    securePort: 443
++    instances: 1
++    sslCertificateRef:
++    caBundleRef:
++    annotations:
++    labels:
++    # [OPTIONAL] if not set, will be obtained from spec.gateway.resources
++    resources:
++      limits:
++        memory: "1024Mi"
```

Comment:
The admin ops instance should have much smaller memory/cpu requirements, right? The admin instances won't serve normal s3 traffic, right?

Reply:
In the vein of what Travis is saying, I think a gig of memory and 1/2 a CPU core seems like it might be more than what this daemon really needs. Those values aren't outlandish, but I would hope they could be lower. 

@arttor do you have CPU/mem data from non-Rook installs that use separate admin daemons to suggest here as a starting point? Or will we have to find recommendations after real world feedback?

Reply:
i've asked colleagues and they advised to use the same resouce limits as for normal RGW installation. Idle resource consumption will be the same and the rest also depends on traffic. So if admin-ops will be used exclusively by Rook then it is makes sense to lower resouce limits. But other customers may integrate admin-ops useage/quota api to 3rd party systems to manage billing and quota. If usage/quota is scraped per bucket then it can create high load when number of buckets is high.